dotCMS/src/main/java/com/dotcms/graphql/exception/CustomGraphQLException.java

```diff
@@ -0,0 +1,39 @@
+package com.dotcms.graphql.exception;
+
+import graphql.ErrorType;
+import graphql.GraphQLError;
+import graphql.language.SourceLocation;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CustomGraphQLException extends RuntimeException implements GraphQLError {
+    private final int errorCode;
+
+    public CustomGraphQLException(int errorCode, String errorMessage) {
+        super(errorMessage);
+
+        this.errorCode = errorCode;
+    }
+
+    @Override
```

Comment:
Is the front end prepared to deal with this "extensions" attribute?
I mean is there any particular reason we want to have it ?